fabfile.py

```diff
@@ -1,197 +1,518 @@
+with {1}'
+        raise ValueError(er_temp.format(current_dir, project_beginning))
+    return current_dir.replace(project_beginning, "")
+
 
 @task
-def deploy(key_file_name="../ec2.pem"):
-    env.key_filename = key_file_name
-    changes = local("git status --porcelain", capture=True)
-    if len(changes):
-        print " {}".format(changes)
-        proceed = prompt(
-            "you have uncommited changes, do you want to proceed",
-            default=False,
-            validate=bool
+def deploy_test(backup_name):
+    new_branch = infer_current_branch()
+    _deploy(new_branch, backup_name)
+    new_status = run_management_command("status_report", Env(new_branch))
+    print("=" * 20)
+    print("new environment was")
+    print(new_status)
+    print("=" * 20)
+
+
+def validate_private_settings():
+    private_settings = get_private_settings()
+    if "host_string" not in private_settings:
+        raise ValueError(
+            'we need a host string inorder to scp data to a backup server'
         )
 
-        if not proceed:
-            return
-
-    git_branch_name = local('git rev-parse --abbrev-ref HEAD', capture=True)
-    with prefix(". /usr/share/virtualenvwrapper/virtualenvwrapper.sh"):
-        with prefix("workon {}".format(virtual_env_name)):
-            run("git fetch")
-            run("git checkout {}".format(git_branch_name))
-            run("git pull origin {}".format(git_branch_name))
-            run("pip install -r requirements.txt")
-            run("python manage.py migrate")
-            run("python manage.py collectstatic --noinput")
-            run("supervisorctl -c etc/test.conf restart gunicorn")
-            run("supervisorctl -c etc/test.conf restart celery")
+    if "password" not in private_settings:
+        raise ValueError(
+            'we need the password of the backup server inorder to scp data to a backup server'
+        )
 
 
 @task
-def restart_all(key_file_name="../ec2.pem"):
-    env.key_filename = key_file_name
-    with prefix(". /usr/share/virtualenvwrapper/virtualenvwrapper.sh"):
-        with prefix("workon {}".format(virtual_env_name)):
-            run("supervisorctl -c etc/test.conf restart gunicorn")
-            run("supervisorctl -c etc/test.conf restart celery")
-            run("supervisorctl -c etc/test.conf restart gloss")
-            run("supervisorctl -c etc/test.conf restart gloss_flask")
+def deploy_prod(old_branch):
+    new_branch = infer_current_branch()
+    old_env = Env(old_branch)
+    new_env = Env(new_branch)
+
+    validate_private_settings()
+    dump_str = "sudo -u postgres pg_dump {0} -U postgres > {1}"
+    local(dump_str.format(old_env.database_name, old_env.release_backup_name))
+
+    cron_write_backup(new_env)
+    cron_copy_backup(new_env)
+    old_status = run_management_command("status_report", old_env)
+    _deploy(new_branch, old_env.release_backup_name)
+    new_status = run_management_command("status_report", new_env)
+
+    print("=" * 20)
```

Comment:
Does this mean that we're expected to eyeball these?
Can't we have a compare function so that a computer can test for equivalence? :)